src/skeleplex/measurements/branches.py

```diff
@@ -0,0 +1,472 @@
+import logging  # noqa
+import os
+import networkx as nx
+import h5py
+import numpy as np
+import skimage as ski
+import torch
+from tqdm import tqdm
+import concurrent.futures
+
+
+from sam2.build_sam import build_sam2
+from sam2.sam2_image_predictor import SAM2ImagePredictor
```

Comment:
We should add sam2 as an optional dependency. Maybe under the torch group?

Reply:
I added a optional dependency in the other PR... Moved it over here, I added it to an extra group, because SAM is not on pip and requires manual installation of the weights, so should be a conciouse process.